doc/release-notes.md

```diff
@@ -4,3 +4,40 @@ release-notes at release time)
 Notable changes
 ===============
 
+Disabling old Sprout proofs
+---------------------------
+
+As part of our ongoing work to clean up the codebase and minimise the security
+surface of `zcashd`, we are removing `libsnark` from the codebase, and dropping
+support for creating and verifying old Sprout proofs. Funds stored in Sprout
+addresses are not affected, as they are spent using the hybrid Sprout circuit
+(built using `bellman`) that was deployed during the Sapling network upgrade.
+
+This change has several implications:
+
+- `zcashd` no longer verifies old Sprout proofs, and will instead assume they
+  are valid. This has a minor implication for nodes: during initial block
+  download, an adversary could feed the node fake blocks containing invalid old
+  Sprout proofs, and the node would accept the fake chain as valid. However,
+  `zcashd` internally contains checkpoints after Sapling activation for both
+  block heights and cumulative chain work, and does not exit the initial block
+  download phase until the active chain contains at least as much work as the
+  checkpointed chain work. The node would therefore be non-functional (and would
```

Comment:
Hmm... I've looked more closely at the use of `IsInitialBlockDownload`, and being in that mode disables:
- Various parts of the network protocol (e.g. advertising a listening address, broadcasting our block inventory, responding to `getheaders`)
- Mining
- Re-broadcasting unmined wallet transactions
- Sprout-to-Sapling migration

Is this sufficiently non-functional, or should we also implement #3996 as well?